grpc-api/src/main/java/io/deephaven/grpc_api/barrage/BarrageStreamGenerator.java

```diff
@@ -427,6 +430,119 @@ private static int createByteVector(final FlatBufferBuilder builder, final byte[
         return builder.endVector();
     }
 
+    /**
+     * Returns an InputStream of the message in a DoGet arrow compatible format.
+     * @return an InputStream ready to be drained by GRPC
+     */
+    public InputStream getDoGetInputStream(final SubView view) throws IOException {
```

Comment:
I think it would be reasonable to assert that all columns are adds, no columns are modes, etc, and simplify the code accordingly.